scripts/language_model/word_language_model.py

```diff
@@ -354,30 +357,51 @@ def train():
             data_list = gluon.utils.split_and_load(data, context, batch_axis=1, even_split=True)
             target_list = gluon.utils.split_and_load(target, context, batch_axis=1, even_split=True)
             hiddens = detach(hiddens)
+            model.collect_params().zero_grad()
             Ls = []
             L = 0
             with autograd.record():
                 for j, (X, y, h) in enumerate(zip(data_list, target_list, hiddens)):
-                    output, h, encoder_hs, dropped_encoder_hs = model.forward(X, h)
+                    output, h, encoder_hs, dropped_encoder_hs = model(X, h)
                     l = joint_loss(output, y, encoder_hs, dropped_encoder_hs)
                     L = L + l.as_in_context(context[0]) / X.size
-                    Ls.append(l/X.size)
+                    Ls.append(l / X.size)
                     hiddens[j] = h
             L.backward()
             grads = [p.grad(d.context) for p in parameters for d in data_list]
             gluon.utils.clip_global_norm(grads, args.clip)
 
+            param_dict_batch_i = {k: v.data(context[0]).copy()
+                                  for k, v in model.collect_params().items()}
             trainer.step(1)
 
+            alpha = 1.0 / max(1, batch_i - avg_trigger + 1)
```

Comment:
With my other comments, we should add 2 here. You can think about the case where we only perform 2 iterations, and final avg_trigger is 0, and the output is the average of these two iterates.

Reply:
In the 2 iterations case, the `alpha = 1 / max(1, 1-0+1) = 1/2`, the output will equal to the average of the first two iterations, it seems to be correct to me.

Reply:
ur batch_i starts from 0. It was 1 iteration. I have updated my comment.

Reply:
the two iterates refer to w_0 and w_1

Reply:
in iteration 0, `param_dict_avg = w0`, in iteration 1, `param_dict_avg = (w0+w1)/2` , seems to be correct?

Reply:
Running the model for only one iteration means only one backward is performed. The routine you mention indicates 2 iterations.